src/deluge/gui/views/arranger_view.cpp

```diff
@@ -2494,34 +2494,9 @@ void ArrangerView::selectEncoderAction(int8_t offset) {
 			                      clipInstance->clip, NULL);
 		}
 
-		int32_t newLength;
-
-		// No newClip means this will become a white clip
-		if (!newClip) {
-			// which will have the same length as the original clip instance
-			newLength = desiredLength;
-		}
-
-		else {
-			// choosing a section clip will reset the clip instance length to the length of that section clip
-			newLength = newClip->loopLength;
-		}
-
-		// Make sure it's not too long
-		ClipInstance* nextClipInstance = output->clipInstances.getElement(pressedClipInstanceIndex + 1);
-		if (nextClipInstance) {
-			int32_t maxLength = nextClipInstance->pos - clipInstance->pos;
-
-			if (newLength > maxLength) {
-				newLength = maxLength;
-			}
-		}
-		if (newLength > kMaxSequenceLength - clipInstance->pos) {
-			newLength = kMaxSequenceLength - clipInstance->pos;
-		}
 		// log action
 		Action* action = actionLogger.getNewAction(ActionType::CLIP_INSTANCE_EDIT, ActionAddition::ALLOWED);
-		clipInstance->change(action, output, clipInstance->pos, newLength, newClip);
+		clipInstance->change(action, output, clipInstance->pos, desiredLength, newClip);
```

Comment:
Should we preserve extending the length if the new clip is longer than the old one?

Reply:
When editing an existing clip instance it makes sense to me to always keep it the same length. In the context of an arrangement it makes sense for the intention to be: in this specific range of the arrangement, I want to change what happens, no matter the length of the source clip.

When creating a new clip instance, it does make more sense to have the result be the length of the chosen clip. This also gives the user a very quick choice between keeping an existing clip instance the same length (hold + turn select) and changing the instance length to clip length (tap to delete, hold + turn select to create new clip instance at clip length)

I'll implement this

Reply:
Yeah but I think this code also covers the initial clip selection, like if you put in a 4 bar blue clip (e.g. last clip you used) then try to change it to your 8 bar pink clip